src/corner-algorithms.jl

```diff
@@ -145,3 +145,43 @@ function gammacovs(cov_xx, cov_xy, cov_yy, gamma::Float64 = 1.4)
 
     filt_cov_xx, filt_cov_xy, filt_cov_yy
 end
+
+
+"""
+```
+moravec_response = moravec(img; [threshold], [window_size])
+```
+Compute the corner response values for each pixel in the input image using the Moravec corner detection algorithm.
+
+    # Arguments
+    - `img::AbstractArray`: The input image.
+    - `window_size::Int`: Size of the window for calculating the corner response.
+    
+    # Keyword Arguments
+    - `args...`: Additional arguments passed to the `gradcovs` function.
+    # Resources
+    - https://vincmazet.github.io/bip/detection/corners.html
+    - http://www0.cs.ucl.ac.uk/staff/g.brostow/classes/IP2008/L7_CornerDetection.pdf
+
+"""
+
+function moravec(img::AbstractArray; window_size::Int = 3, args...)
+    gradient_x, gradient_y = gradcovs(img, args...)
+    
+    corner_response = similar(img, Float64)  # Initialize a corner response image
+
+    for y in 0:size(img, 2) - window_size + 1
+        for x in 0:size(img, 2) - window_size + 1
+            min_sum_diff = Inf
+            for dy in -1:1
+                for dx in -1:1
+                    sum_diff = sum((gradient_x[y+1:y+window_size-2, x+1:x+window_size-2] .- gradient_x[y+dy:y+dy+window_size-2, x+dx:x+dx+window_size-2]).^2)
+                    min_sum_diff = min(min_sum_diff, sum_diff)
+                end
+            end
+            corner_response[y, x] = min_sum_diff
+        end
+    end
```

Comment:
This is gonna end up looking more like this:
```suggestion
    for y in 2*window_size+1 : size(img, 1) - 2*window_size + 1
        for x in 2*window_size+1 : size(img, 2) - 2*window_size + 1
            min_sum_diff = Inf
            for dy in -drange:drange
                for dx in -drange:drange
                    if(dy == 0 && dx == 0) continue end
                    sum_diff = sum((img[y - half : y + half, x - half : x + half] .- img[y - half + dy : y + half + dy, x - half + dx : x + half + dx]).^2)
                    min_sum_diff = min(min_sum_diff, sum_diff)
                end
            end
            corner_response[y, x] = min_sum_diff
        end
    end
```

Notice:
- How we iterate, in terms of x and y ranges
- We don't want to use dx, dy being 0 as it's essential same as initial subimage which results in 0 only.
- You can make -1:1 to be flexible by allowing it be a keyword argument `drange`